src/training/sft.py

```diff
@@ -158,7 +161,39 @@ def save_model(self) -> None:
         else:
             self.accelerator.save_state(path)
 
-    def step(self, batch: dict) -> None:
+    def warmup_step(self) -> None:
+        '''Preallocates memory for maximum sequence length in order to avoid nasty OOM shocks'''
+        model_config = self.model.gpt_neox.config
```

Comment:
Needs to be:

```suggestion
        model_config = self.model.module.gpt_neox.config
```

Because of DeepSpeed.